app/views/session_timeout/_expire_session.js.erb

```diff
@@ -2,6 +2,10 @@ var sessionTimeoutIn = <%= session_timeout_in %> * 1000;
 var modal = "<%= j render('session_timeout/expired') %>";
 
 function displaySessionTimeoutModal() {
+  var inputs = document.querySelectorAll('input[type="submit"]');
+  inputs.forEach(function(element) {
```

Comment:
I don't think you can directly loop over the results of querySelectorAll since it's not a regular array.  That could be the reason for the `TypeError: undefined is not a constructor (evaluating 'inputs.forEach(function(element) {`.  I tried this instead and it seems to do the trick:
```
  var inputs = document.querySelectorAll('input[type="submit"]'), i;
  for (i = 0; i < inputs.length; ++i) {
    inputs[i].disabled = true;
  }
```